### PR TITLE
return a new auth token for a valid config token

### PR DESCRIPTION
**Pull request by ericcf**
[#87064660]

### CODE REVIEW

File: app/controllers/api/authentication_tokens_controller.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  class AuthenticationTokensController < ActionController::Base
+    RESOURCE_TYPE = "authenticationTokens"
+
+    include Concerns::CorsSettings
+
+    after_action :cors_set_access_control_headers
+
+    def create
+      if configuration_token && authentication_token
+        render(json: {
```
Comment (michaelwehrley) [inline app/controllers/api/authentication_tokens_controller.rb]: JSON API structure! :hamburger:

File: app/models/configuration_token.rb [hunk 2]
```diff
@@ -12,6 +12,29 @@ class ConfigurationToken < ActiveRecord::Base
...
+  # Returns case insensitive match.
+  def self.find_match(value)
+    return nil unless value.is_a?(String) && value.length == TOKEN_LENGTH
+
+    where(arel_table[:expires_at].gt(DateTime.current))
+      .where(arel_table[:value].matches(value))
+      .first
+  end
+
+  def make_authentication_token(client_uuid:)
```
Comment (michaelwehrley) [inline app/models/configuration_token.rb]: did you mean to write `client_uuid:` or `client_uuid`?

Comment (ericcf) [inline app/models/configuration_token.rb]: The former. See this: https://robots.thoughtbot.com/ruby-2-keyword-arguments

File: spec/models/authentication_token_spec.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    { participant: Participant.first, client_uuid: 321 }
+  end
+
+  describe "validations" do
+    it "accepts valid attributes" do
+      AuthenticationToken.create!(valid_attributes)
+    end
+  end
+
+  describe "initialization" do
```
Comment (michaelwehrley) [inline spec/models/authentication_token_spec.rb]: I like how you checked length of both the `value` and `uuid`.  Adding "length" as a constant of the class not only helps in the creation, but also here in testing.

**Comment by michaelwehrley** [comment]
:+1: :black_joker:

**Pull request merged**